gtfs-realtime/spec/en/reference.md

```diff
@@ -146,10 +161,10 @@ The update is linked to a specific stop either through stop_sequence or stop_id,
 
 | _**Field Name**_ | _**Type**_ | _**Cardinality**_ | _**Description**_ |
 |------------------|------------|-------------------|-------------------|
-| **stop_sequence** | [uint32](https://developers.google.com/protocol-buffers/docs/proto#scalar) | optional | Must be the same as in stop_times.txt in the corresponding GTFS feed. |
-| **stop_id** | [string](https://developers.google.com/protocol-buffers/docs/proto#scalar) | optional | Must be the same as in stops.txt in the corresponding GTFS feed. |
-| **arrival** | [StopTimeEvent](#message-stoptimeevent) | optional |
-| **departure** | [StopTimeEvent](#message-stoptimeevent) | optional |
+| **stop_sequence** | [uint32](https://developers.google.com/protocol-buffers/docs/proto#scalar) | conditionally required | Must be the same as in stop_times.txt in the corresponding GTFS feed.  Either stop_sequence or stop_id must be provided within a StopTimeUpdate - both fields cannot be empty.  stop_sequence is required for trips that visit the same stop_id more than once (e.g., a loop) to disambiguate which stop the prediction is for. |
+| **stop_id** | [string](https://developers.google.com/protocol-buffers/docs/proto#scalar) | conditionally required | Must be the same as in stops.txt in the corresponding GTFS feed. Either stop_sequence or stop_id must be provided within a StopTimeUpdate - both fields cannot be empty. |
+| **arrival** | [StopTimeEvent](#message-stoptimeevent) | conditionally required | Either arrival or departure must be provided within a StopTimeUpdate - both fields cannot be empty. |
```

Comment:
@jxeeno Good catch again!  Yes, for `SKIPPED` neither `arrival` nor `departures` need to be provided ([E043](https://github.com/CUTR-at-USF/gtfs-realtime-validator/blob/master/RULES.md#e043---stop_time_update-doesnt-have-arrival-or-departure) for our validator).  Also, it's actually an error to provide `arrival` and `departure` for `ScheduleRelationship` of `NO_DATA` ([E042](https://github.com/CUTR-at-USF/gtfs-realtime-validator/blob/master/RULES.md#e042---arrival-or-departure-provided-for-no_data-stop_time_update)).  I'll fix both these items.